crates/router/src/core/payments.rs

```diff
@@ -1082,7 +1082,7 @@ where
             .attach_printable("Invalid straight through algorithm format in payment attempt")?,
     };
 
-    let request_straight_through: Option<api::RoutingAlgorithm> = request_straight_through
+    let request_straight_through: Option<api::StraightThroughAlgorithm> = request_straight_through
```

Comment:
In line 1086  should it be parse_value("StraightThroughAlgorithm").

Reply:
Nice catch. Fixed.